### PR TITLE
[ADVAPI32][SHELL32][DESK.CPL] Kill copy-paste of RegLoadMUIString.

### PR DESCRIPTION
**Pull request by Getequ**
RegLoadMUIStringW/A is part of advapi32. shell32 and desk.cpl has contained exact line-by-line copy of that function

### CODE REVIEW

File: dll/win32/advapi32/advapi32.spec [hunk 1]
```diff
@@ -682,3 +682,5 @@
 682 stub WmiSetSingleItemW
 683 stub Wow64Win32ApiEntry
 684 stdcall WriteEncryptedFileRaw(ptr ptr ptr)
+685 stdcall RegLoadMUIStringW(ptr wstr wstr long ptr long wstr)
```
Comment (ThFabba) [inline dll/win32/advapi32/advapi32.spec]: This is not okay. Our imports must match Win2003.
If our desk.cpl and shell32.dll import these functions, they will no longer work on Win2003.

Comment (HBelusca) [reply dll/win32/advapi32/advapi32.spec]: Don't we have "advapi32_vista.dll" ?

Comment (Getequ) [reply dll/win32/advapi32/advapi32.spec]: But shell32.dll even now have this 
remove_definitions(-D_WIN32_WINNT=0x502)
remove_definitions(-D_WIN32_WINNT=0x600)

As well as 60+ other DLLs. Some of them redefine this config to x601..x603. It assumed that thay can not work as x502, isn't it?

Comment (Getequ) [reply dll/win32/advapi32/advapi32.spec]: Arghhh, these xxx_vista.dll even shitter than this)

Comment (Getequ) [reply dll/win32/advapi32/advapi32.spec]: @ThFabba  What if i try to wrap this copy-paste into #ifdef < 0x600  ?

Comment (ThFabba) [reply dll/win32/advapi32/advapi32.spec]: Yes, moving it to advapi32_vista would work to de-duplicate the code. That dll can be copied to Win2003 so it doesn't break this method of testing.

Wrapping the duplicated copies in an ifdef seems acceptable, though I'm not sure what it achieves. If your goal is just to fix build with that define, I guess that works.

**Comment by Getequ** [comment]
@ThFabba  hi, i moved code to advapi32_vista and linked all three advapi32 shell32 and desk to this dll

File: dll/win32/advapi32/advapi32.spec [hunk 2]
```diff
@@ -682,3 +682,5 @@
 682 stub WmiSetSingleItemW
 683 stub Wow64Win32ApiEntry
 684 stdcall WriteEncryptedFileRaw(ptr ptr ptr)
+685 stdcall RegLoadMUIStringW(ptr wstr wstr long ptr long wstr) advapi32_vista.RegLoadMUIStringW
```
[the remark this answers is not in the document]
Comment (Getequ) [reply dll/win32/advapi32/advapi32.spec]: I wrapped them under -version=0x600+ condition

File: dll/win32/advapi32_vista/RegLoadMUIString.c [hunk 3]
```diff
@@ -98,7 +71,7 @@ RegLoadMUIStringW(IN HKEY hKey,
 
     if (pszDirectory && *pszDirectory)
     {
-        FIXME("BaseDir parameter not yet supported!\n");
+        //FIXME("BaseDir parameter not yet supported!\n");
```
Comment (learn-more) [inline dll/win32/advapi32_vista/RegLoadMUIString.c]: Why is this commented?

**Comment by JoachimHenze** [comment]
This PR introduced regression CORE-15412, really for no good reason.
This cpl was perfectly standalone before!

**Comment by Getequ** [comment]
Let's copy-paste this function back?

**Pull request merged**